test/Data/Source/Spec.hs

```diff
@@ -25,11 +27,13 @@ prop desc f
 testTree :: Tasty.TestTree
 testTree = Tasty.testGroup "Data.Source"
   [ Tasty.testGroup "sourceLineRanges"
-    [ prop "produces 1 more range than there are newlines" $
-      \ source -> length (sourceLineRanges source) === succ (Text.count "\n" (toText source))
+    [ QC.testProperty "produces 1 more range than there are newlines" $
+      \ source -> QC.label (summarize source) $
+        length (sourceLineRanges source) QC.=== length (Text.splitOn "\r\n" (toText source) >>= Text.splitOn "\r" >>= Text.splitOn "\n")
```

Comment:
Ironically, this implementation is approximately as efficient as `newlineIndices` 😭